src/firewall/functions.py

```diff
@@ -101,6 +102,95 @@ def addr_family_str(family):
 ###############################################################################
 
 
+def _parse_check_fcn(parse_fcn):
+    def f(*a, **kw):
+        try:
+            parse_fcn(*a, **kw)
+        except ValueError:
+            return False
+        return True
+
+    return f
+
+
+def _parse_norm_fcn(parse_fcn, unparse_fcn):
+    def f(*a, **kw):
+        detail = parse_fcn(*a, **kw)
+        return unparse_fcn(*detail)
+
+    return f
+
+
+###############################################################################
+
+
+class EntryType:
+    class ParseFlags(enum.IntFlag):
+        NO_IP6_BRACKETS = 0x1
+
+    def __init__(self, name, fcn_parse, fcn_unparse):
+        self.name = name
+        self.parse = fcn_parse
+        self.unparse = fcn_unparse
+        self.norm = lambda *a, **kw: self._norm(*a, **kw)
+        self.check = lambda *a, **kw: self._check(*a, **kw)
```

Comment:
Why expose them this way? Just rename `_norm()` to `normalize()`.

Reply:
changed for norm (that is, the instance method `_norm()` was renamed to `norm()` and `self.norm = ...` is gone).

However, there is still `self.check = self._check`. The reason is that `EntryType` already has a `@staticmethod` `EntryType.check()`.

In general, there are instance methods `self.{check,parse}()` and static methods `EntryType.{check,parse}()`. They do something different, but their names are still fitting in both cases. Hence the overlapping names and the `self.check = self._check` workaround.